### PR TITLE
Add notification menu item

### PR DESCRIPTION
**Pull request by peterkeating**
Adds a new menu item that displays a count obtained via a query alongside the menu item label that links to a content item on the site.

### CODE REVIEW

File: Migrations/update4.recipe.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+              "Name": "LinkVisualPart",
+              "Settings": {
+                "ContentTypePartSettings": {
+                  "Position": "2"
+                }
+              }
+            }
+          ]
+        }
+      ],
```
Comment (Newnab) [inline Migrations/update4.recipe.json:45]: The LinkVisualPart being on this creates an interesting scenario if they try to turn it into e.g a button... Not anything you need to fix, just an observation, thinking out loud :)

Comment (peterkeating) [reply Migrations/update4.recipe.json:45]: The theme will need to cater for the fact that `notification` could be a button.

File: Views/MenuItemLink-NotificationMenuItem.cshtml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    try
+    {
+        notificationCount = (await Orchard.ContentQueryAsync(query)).Count();
+    }
+    catch
+    {
+        notificationCount = 0;
+    }
+
+    var cssClasses = "notification";
```
Comment (Newnab) [inline Views/MenuItemLink-NotificationMenuItem.cshtml:27]: I'm wondering if this is too generic/vague. Would it be clearer to have `notification-menu-item`? Or give all menu items a `menu-item` class and add `menu-item--with-notification` to this? Something like that?

**Review by Newnab — APPROVED** [review]
Made a couple of comments but nothing is a showstopper.

File: Views/MenuItemLink-NotificationMenuItem.cshtml (hunk 2 above)
Comment (peterkeating) [reply Views/MenuItemLink-NotificationMenuItem.cshtml:27]: I think we would do that in `MenuItem.liquid`.

https://github.com/EtchUK/Etch.OrchardCore.Menu/blob/main/Views/MenuItem.liquid

Comment (peterkeating) [reply Views/MenuItemLink-NotificationMenuItem.cshtml:27]: Updated the `MenuItem` template to add a `nav__item--content-type` modifier.

**Pull request merged**